Source/WebGPU/WebGPU/Queue.mm

```diff
@@ -692,7 +692,9 @@
     // FIXME(PERFORMANCE): Should this temporary buffer really be shared?
     auto newBufferSize = static_cast<NSUInteger>(dataByteSize);
```

Comment:
instead of changing lines 695..697 can we just say:
```
auto newBufferSize = static_cast<NSUInteger>(dataByteSize) - dataLayout.offset;
```
here?

Otherwise the `bool noCopy = newBufferSize >= largeBufferSize;` condition doesn't make as much sense.